### DISCUSSION TITLE
script hangs at kitty command

### DISCUSSION THREAD

**Discussion by Ciflire** [kovidgoyal/kitty#8052] [Q&A]
Hi i'm trying to make a script to re-render puml and display it on kitty with this script
```bash

inotifywait \
	-m puml/ \
	--recursive \
	--event close_write ./puml | while read -r directory event file; do
	echo "loop"
	plantuml -progress "$PWD"/puml/"${file%.*}".puml -o "$PWD"/images/
	kitten icat --transfer-mode memory --clear "$PWD"/images/"${file%.*}".png
	echo "loop end"
done
```
However that script hangs at the `kitten icat` command, i'm probably misusing it or misunderstanding something here, trying to see how to solve this problem

**Comment by kovidgoyal** [comment]
probably because there is no tty device available in whatever context your script is running in.  Read https://sw.kovidgoyal.net/kitty/kittens/icat/  to see how to use it in such contexts.

**Reply by Ciflire** [reply]
i have read the doc, however, it outputs the image in kitty, but doesn't end it's execution for some reason (or notify it), so the script hangs at the kitty line
the following scripts work, and output what i expect them to but not the one i wrote
```bash
while [ 0 -le 1 ] 
do
echo 'begin loop'
kitten icat images/diag_seq1.png
echo 'end loop'
done
```

```bash
inotifywait -m puml --recursive --event close_write | while read -r directory event file
do
echo $file
done
```

so i'm a bit confused with what's happening here
